### PR TITLE
feat: Add reply editor mode to the store

### PR DESCRIPTION
**Pull request by muhsin-k**
In order to implement more AI options, we need currently selected reply mode in the command bar and other AI assistance-related components. This pull request adds the ability to store reply mode to the store.

### CODE REVIEW

**Review by nithindavid — COMMENTED** [review]
Code looks good, I'll test this tomorrow.

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 1]
```diff
@@ -781,6 +781,9 @@ export default {
     },
     setReplyMode(mode = REPLY_EDITOR_MODES.REPLY) {
       const { can_reply: canReply } = this.currentChat;
+      this.$store.dispatch('draftMessages/setReplyMEditorMode', {
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:784]: is the `M` in setReply**M**EditorMode intentional, feels like a typo

**Review by scmmishra — APPROVED** [review]

**Comment by scmmishra** [comment]
Tested locally, LGTM!

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue (hunk 1 above)
Comment (muhsin-k) [reply app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue]: Yes. Fixed.

**Pull request merged**